crypto/params.c

```diff
@@ -408,7 +408,7 @@ int OSSL_PARAM_get_int64(const OSSL_PARAM *p, int64_t *val)
         switch (p->data_size) {
         case sizeof(double):
             d = *(const double *)p->data;
-            if (d >= INT64_MIN && d <= INT64_MAX && d == (int64_t)d) {
+            if (d >= INT64_MIN && d == (int64_t)d) {
```

Comment:
The check against INT64_MIN is the same -- it just happens that INT64_MIN can be represented as a double.

Reply:
The check for d being integral is already done, so `d >= ldexp(1, -64) && d < ldexp(1, 64)` would work here but I don't think we can handle the libm dependency.

This is a bit obtuse, but I think `d >= INT64_MIN && -d > INT64_MIN` will work.